py3status/modules/wwan.py

```diff
@@ -322,6 +342,41 @@ def _get_bearer(self, modem_proxy):
             pass
         return bearer
 
+    def _get_messages(self, modem_proxy):
+        messages_data = {}
+        key = 0
+        try:
+            messages = modem_proxy.Messages
+            # loop on sms list for organize them
+            sms_data = {}
+            for msg in messages:
+                id = int(msg.replace(STRING_MODEMMANAGER_SMS_PATH, ''))
+                sms_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, msg)
+                contact = sms_proxy.SMSC
+                text = sms_proxy.Text
+                key += 1
+                sms_data[key] = {'id': id, 'contact': contact, 'text': text}
+                messages_data['last_contact'] = '{}'.format(contact)
+                messages_data['last_text'] = '{}'.format(text)
```

Comment:
The whole area here looks incorrect. We print `last_*` because we're looping through everything and print the `last_*` thing in the loop. If we want everything in the loop, we need a list `new_data = []` to store things like this:
```
        new_data.append(self.py3.safe_format(self.format_message, {
            'id': id, 'contact': contact, 'text': text})
        )
```

See module `coin_market` for an example. In this list is a list of composites. Out there in the return, you give it to `format_message` along with a counter for `message` too.